### ISSUE TITLE
Nelson Time Tracking

### ISSUE THREAD

**Issue by markwilliamfirth** [dwyl/hq#226]
Nelson to start tracking time on projects from 13th February onwards utilising [Focus Time](http://focustimeapp.com/) or another equivalent app. Tracking will split time into learning Elixir, QA, answering questions, project work as well as other categories that Nelson deems suitable.

**Comment by nelsonic** [comment]
Thanks @markwilliamfirth 👍

**Comment by nelsonic** [comment]
@markwilliamfirth / @iteles  and who ever `else` is _curious_/_interested_, this is how I spent last week:
![image](https://cloud.githubusercontent.com/assets/194400/23090950/469e2732-f5a2-11e6-9d4f-8293c572a624.png)
Note: I think there's a _bug_ in the "***Be Focussed***" app because I actually have quite a few Tasks marked as "done" for the week!
![image](https://cloud.githubusercontent.com/assets/194400/23091020/0c3b8cf4-f5a4-11e6-9220-8c85043cf6a4.png)
Maybe I need to _pay_ for the feature of it actually displaying the tasks that I completed in the report ...
Or maybe I could _fix_ the feature in 5 minutes if the code was Open Source.
Anyway ... here's the "_breakdown_" by day of the week:
![image](https://cloud.githubusercontent.com/assets/194400/23090954/53a31c12-f5a2-11e6-8415-5a1d1eff04ce.png)
![image](https://cloud.githubusercontent.com/assets/194400/23090956/64e7f0e2-f5a2-11e6-97c0-eccb4751f285.png)
![image](https://cloud.githubusercontent.com/assets/194400/23090958/6f331f86-f5a2-11e6-847b-8b55646773ea.png)
![image](https://cloud.githubusercontent.com/assets/194400/23090964/79f62b34-f5a2-11e6-991f-e1c5a489121a.png)
![image](https://cloud.githubusercontent.com/assets/194400/23090965/881dd374-f5a2-11e6-9e83-357318c4a60c.png)

> More detail review of the **Focus Timer** over on: https://github.com/dwyl/time/issues/184

# tl;dr

The `main` _reason_ I _want_ to track my own work/activity is _not_ so that `i` know what I've done or what I'm currently doing (_I know **exactly** what I'm doing at **all times** and have **no trouble focussing** on getting my work `done`... provided people aren't making [**noise**](https://github.com/dwyl/hq/issues/113) around me or constantly interrupting me ..._) 
The _reason_ I want to track what I'm doing is so that it's clear to everyone `else` when I'm _available_ for other tasks/discussions.

> Most people have not _invested_ the time to _understand_ how _other_ people work _best_.

Most people don't realise that there are people who ***cannot*** get work done with noise.
see: https://www.amazon.co.uk/Quiet-Power-Introverts-World-Talking/dp/0141029196
The _reason_ why "***QuietSpace***" will have a _waiting list_ for membership is that there are ***many*** people who cannot focus when there is noise. `.some()` of those people are the most creative/productive people on the planet. giving them a _haven_ where they can focus is a USP you can take to the *bank*! we need to _fiercely protect_ their "***quiet time***" and not be "_afraid_" to tell others to ["_please be quiet so other people who are more noise-sensitive can get their work done..._"](https://youtu.be/GqvP1DZQyRU?t=1m12s)


> The _reason_ I wanted to build `Time` was so that I could _communicate_ with people on the _team_. It's not about "**time sheets**" or "***billable hours***" (_though those things are **useful** when running a company of **any** size that has to invoice for time!_) it's about ***communication***! **Focus Timer** does _nothing_ to help me _communicate_, it does _nothing_ to solve the _real_ problem: _constant interruptions_. 

The first time I thought of the "Time" App was _many_ years ago when watching employees "clock-in" and "clock-out" at a factory I worked at. It struck me as incredibly "_crude_" way of tracking activity. Also people had a "_system_" whereby they would "_cover_" for each other when someone had to leave early one of their colleagues would clock them out when they left so the early-leaver would still get paid for the hours (_even though they left early for whatever reason..._)

When @iteles and I worked (_so we could pay our own way through Uni and finish debt-free..._) as "temps" (AKA "zero hours wage slaves"_) for  http://www.timerecruitment.co.uk (_loving those **stock photos**!_) we were required to submit time sheets but weren't given _any_ way of _tracking_ our activity. we just wrote it down on paper and faxed it in at the end of the week. If we had an ***app*** we could have helped _hundreds_ of "temp" people effectively track their time. And guess what those apps _exist_ now other people/companies _built_ them `while` we were stuck working for "_the man_" instead of working on our own thing ... and it _kills_ me inside when an idea is _implemented_ (_poorly!_) by someone `else` ...

When I worked at [***Exploitte***](http://www.urbandictionary.com/define.php?term=Exploitte.) we were _required_ to fill in "Timesheets" in "SAP" broken down in **15 min** increments. 
Did they give us a way of tiying the ***work*** to the timesheets? ***heck no***!
The system for getting work done (_a bunch of MS Excel macros strung together with some VB UI that cost "The Firm" 4 years of dev & $250M! and was barely "fit-for-purpose"_) was in _no way_ integrated with the time-reporting system which was developed by a totally different (***closed source***) "vendor"...
That data went into a payroll _black hole_ and wasn't used to help people _learn_ how to be more effective or to _communicate_ how much _actual_ time was being spent on an activity that some "_Senior Manager_" said "_should only take 5 minutes_" (_but actually took a whole weekend!_)

> Memory-lane musings and nostalgia aside, the **need** for accurate, painless and passive time/activity measurement to aid ***personal effectiveness*** is a ***very real need*** and anyone who does not _immediately_ ***see*** that need has not worked in enough places to see the _breathtaking inefficiency_ many people/companies have!
And when I say that this is a "***Billion Dollar*** idea" I'm not "_kidding_", _anyone_ who knows the history of computing knows that one of the pre-cursor or "founding" companies of IBM was a company called "***International Time Recording Company***" i.e. the first company to offer time Recording in factories.
For the uninitiated, see: https://en.wikipedia.org/wiki/History_of_IBM#Chronology
and... helping companies track their "_employees_" is still one of IBMs core "_cash cow_" applications!
I don't want to help the **Company** track it's people I want to help the people track themselves and ***communicate*** with their peers so that the whole _team_ can be more effective!

I've worked at _enough_ organization where people just **show up**, **_pretend_ to work** (_mostly because they were **demotivated** by the **boring** tasks being handed down by "PMs" who themselves were **disinterested** in the work! not because of any lack of ability/skill!_), collect a paycheck 
`.then` go home and _vegetate_ on the couch watching TV or play video games to numb their minds, or _worse_ ... yes, some of the smartest people use substances to ***not think***! 
And that my friends is one of the _saddest_ things to witness ...  
Highly intelligent people who have no ***purpose*** in life and use something to "***switch off***" from their reality ... some of the most [**exceptionally talented people**](# "I was going to link to someone but decided not to because this isn't about any one person, the problem is pervasive!!") we know spend large sums of cash on flights to Ibiza and "Mandy" because their lives are _meaningless_. (_no offense to anyone..._) 
and it's _only_ going to get [_worse_](https://www.theguardian.com/technology/2017/jan/17/facebook-mark-zuckerberg-testifies-oculus-rift-lawsuit) with improvements in VR technology ... [`#isThisLiving?`](https://youtu.be/FHCYHldJi_g)

Unless we @dwyl ***do something*** about it ... 

So, who's _with_ me? 
Who has _felt_ the pain of working in a team where "_status updates_" were the (only) way to _communicate_?
Who thinks there is (_has to be!_) a ***better way*** of communicating in teams?
Who is going to [***learn***](https://github.com/search?q=org%3Adwyl+learn) what we need to _know_ ...
so we can _build_ what we ***need*** to `make`?!
Who wants help use _convert_ our _pile_ of cash (_from operations not selling out to "investors"_) 
into **Open Source Software** that _anyone_ can use to ***accomplish more*** and help ***live a better life***?! ❤️  ✅ 

This is what I am working on today:
+ https://github.com/issues?q=is%3Aopen+is%3Aissue+assignee%3Anelsonic+label%3Apriority-1

How about ***you***? 😉

**Comment by markwilliamfirth** [comment]
@nelsonic in terms of working hours (I assume the yoga is before work?) split between projects what's the number of hours on each project? Kinda difficult to see from the graph. Did you track interruptions / time spent answering questions people asked you?

**Comment by nelsonic** [comment]
Hi @markwilliamfirth,
Yes, yoga is tracked but not part of work, but I don't know how to _exclude_ it from the graph, sorry.

The "aim" is to have 10 "pomodoros" of _productive_ work per day. 
hence the red dotted-line on the bar chart. 
(_most people don't get **5 hours** of high priority ["**deep**" **work**](https://www.quora.com/How-much-deep-work-are-you-able-to-do-during-the-day) done in any given **day**..._)

If I tracked the interruptions it would be the _biggest_ category. (_I don't currently track interruptions because the tool(s) don't facilitate this tracking ... "oh wait, you're interrupting me, let me start a timer to track how much of my time you're going to waste..."_) 

> One of the features I want to create in Time is that when you `pause` one timer it _automatically_ starts another timer which you can then choose to keep and label as an "interruption" or "conversation with Susan about widgets..." or _discard_ if its not relevant.
To me this is an _obvious_ feature, but for some reason I haven't seen other apps do it.

The number of hours on each project can be derived from the number of pomodoro units.
Each unit is **25 minutes** followed by a 5 min "rest" (_which I occasionally take but most of the time I just work through_) so two pomodoro units is an hour. 
So **8 units** for ***Unbound*** on Friday is 4 hours. 
This is the time I actually spent on the workshop (_because I cannot count the waiting time..._) 
the other time was breaks, chatting with the team and reviewing PRs for other projects (_which I did when everyone else went for lunch because I wanted to maximise my time..._).

Since we are _not_ currently invoicing for our time based on our Pomodoro Timers (_which would be great but we aren't there yet for a **number** of reasons..._) I am not converting my Timer data into billable hours on invoices. When @iteles is _ready_ to start doing invoicing that way, I'm **100% on board**. 

I will `continue` tracking my own time using Focus Timer _until I carve out_ the time to work on https://github.com/dwyl/time ... (_anyone `else` see the irony...?_)

What `else` is required for this issue? (_thanks again for opening it, but I need a clear "acceptance criteria" so I know when I can consider it "done"._) 🤔

**Comment by markwilliamfirth** [comment]
@nelsonic the original issue was because there was an idea that we could try and direct more of your time towards higher priority projects, but we couldn't do so until understanding where your time was being spent first.

You mentioned interruptions were the biggest amount of time spent, but looking at the above it doesn't reflect this assumption.

The below shows estimates of percentages of where you are currently spending your time:
<img width="186" alt="screen shot 2017-02-20 at 13 52 17" src="https://cloud.githubusercontent.com/assets/5732209/23127686/d4329c9a-f773-11e6-87f2-c60db1ee9a4e.png">

The below shows estimates of the company's priorities:
<img width="222" alt="screen shot 2017-02-20 at 13 53 39" src="https://cloud.githubusercontent.com/assets/5732209/23127761/12b637d8-f774-11e6-802c-81cd1b0f824c.png">

The issue is - **how do we assign your time to more closely match the company's priorities?**

**Comment by nelsonic** [comment]
@markwilliamfirth the realities of our _existing_ commitments to client work do not reflect our priorities as a company. but until the client work is delivered we cannot _focus_ our efforts on the highest priority/value activities. I've spent 18h learning Elixir & Phoenix over the weekend so that I _can_ (_eventually_) focus on building the Time app. sadly I have nothing more to show for it than "_progress_" through the "Programming Elixir" or "Programming Phoenix" book(s). 
But I _hope_ to have more to show for my effort(s) soon... 

meanwhile ... in relation to the original requirement of `this` issue, what `else` is needed?

**Comment by markwilliamfirth** [comment]
I guess the issue is about matching time (as much as possible) with priorities and if the above is correct then we need to reassess company priorities to take into account current services and commitments as client work is clearly very important to the company.

**Comment by nelsonic** [comment]
@markwilliamfirth we _need_ the priorities to be well-defined so we _all_ know how to focus our Time (_pun intended_).

**Comment by markwilliamfirth** [comment]
Closing this issue

**Comment by nelsonic** [comment]
@markwilliamfirth I'm happy to keep it open and assigned to me as I'm happy to keep tracking my time/activity with existing tools. just don't think it's a "Priority 1" issue.  👍
